### PR TITLE
Update r0.17 

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*
Updates the r0.17 branch, making changes according to the new infrastructure and CI requirements. Also bumps up tensorflow and pytorch to latest version. Update pinned deps as well.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by AnirudhDagar** [comment]
@terrytangyuan the TF section on dropout seems to be unstable. It failed in [this](http://ci.d2l.ai/blue/organizations/jenkins/d2l-en/detail/PR-2055/1/pipeline) run. But passed when I re-ran the CI. Could you please take a look at this issue? Thanks!

**Comment by AnirudhDagar** [comment]
@astonzhang while building the pytorch pdf, CI fails trying to build the [d2l API section](http://d2l.ai/chapter_appendix-tools-for-deep-learning/d2l.html) for torch.meshgrid. We earlier (in torch v1.8.1) had the following example from their docs:

```python
>>> x = torch.tensor([1, 2, 3])
>>> y = torch.tensor([4, 5, 6])
>>> grid_x, grid_y = torch.meshgrid(x, y)
>>> grid_x
tensor([[1, 1, 1],
        [2, 2, 2],
        [3, 3, 3]])
>>> grid_y
tensor([[4, 5, 6],
        [4, 5, 6],
        [4, 5, 6]])
```

but now with v1.10 ([from docs](https://pytorch.org/docs/1.10/generated/torch.meshgrid.html?highlight=torch%20meshgrid#torch.meshgrid)) the example now plots after meshgrid too.

```python
x = torch.tensor([1, 2, 3])
y = torch.tensor([4, 5, 6])

grid_x, grid_y = torch.meshgrid(x, y, indexing='ij')
grid_x
grid_y

torch.equal(torch.cat(tuple(torch.dstack([grid_x, grid_y]))),
            torch.cartesian_prod(x, y))

import matplotlib.pyplot as plt
xs = torch.linspace(-5, 5, steps=100)
ys = torch.linspace(-5, 5, steps=100)
x, y = torch.meshgrid(xs, ys, indexing='xy')
z = torch.sin(torch.sqrt(x * x + y * y))
ax = plt.axes(projection='3d')
ax.plot_surface(x.numpy(), y.numpy(), z.numpy())
plt.show()
```

This plot then leads to the image meshgrid.png which obviously is missing in our book. I'm attaching grep output below to help explain the issue. Any ideas on how to fix such an issue? Is it fine to add the plot manually to the [static dir](https://github.com/d2l-ai/d2l-en/tree/master/static) so that it is also copied to `/home/d2l-worker/workspace/d2l-en-r0.17/_build/html/_static` during build. This should technically solve the problem but this may not be the best solution.

```bash
(d2l-en-r0.17-0) d2l-worker@<ip>:~/workspace/d2l-en-r0.17$ grep -r "_static/img/meshgrid"
Binary file _build/html/.doctrees/chapter_appendix-tools-for-deep-learning/d2l.doctree matches
Binary file _build/html/.doctrees/environment.pickle matches
_build/html/_modules/torch/functional.html:<span class="sd">        .. image:: ../_static/img/meshgrid.png</span>
_build/html/chapter_appendix-tools-for-deep-learning/d2l.html:<a class="reference internal image-reference" href="_static/img/meshgrid.png"><img alt="_static/img/meshgrid.png" src="_static/img/meshgrid.png" style="width: 512px;" /></a>
Binary file _build/pdf_pytorch/.doctrees/chapter_appendix-tools-for-deep-learning/d2l.doctree matches
Binary file _build/pdf_pytorch/.doctrees/environment.pickle matches
_build/pdf_pytorch/d2l-en.tex:\noindent\sphinxincludegraphics[width=512\sphinxpxdimen]{{_static/img/meshgrid}.png}
_build/pdf_pytorch/d2l-en.fdb_latexmk:  "_static/img/meshgrid.png" 0 -1 0 ""
_build/pdf_pytorch/d2l-en.fdb_latexmk:  "{_static/img/meshgrid}.png" 0 -1 0 ""
_build/pdf_pytorch/d2l-en.log:LaTeX Warning: File `{_static/img/meshgrid}.png' not found on input line 63624.
_build/pdf_pytorch/d2l-en.log:! Unable to load picture or PDF file '_static/img/meshgrid.png'.
_build/pdf_pytorch/d2l-en.log:l.63624 ...inxpxdimen]{{_static/img/meshgrid}.png}
_build/pdf_pytorch/d2l-en.log:l.63624 ...inxpxdimen]{{_static/img/meshgrid}.png}
_build/pdf_pytorch/d2l-en.log:LaTeX Warning: File `{_static/img/meshgrid}.png' not found on input line 63624.
_build/pdf_pytorch/d2l-en.log:l.63624 ...inxpxdimen]{{_static/img/meshgrid}.png}
_build/pdf_pytorch/d2l-en.log:File: _static/img/meshgrid.png Graphic file (type bmp)
_build/pdf_pytorch/d2l-en.log:<_static/img/meshgrid.png>
```

**Comment by astonzhang** [comment]
The current API page shows too many non-d2l-lib members (those not saved to d2l lib via #@save, such as DL framework operators). Can you investigate how to hide these (so torch. meshgrid will be hidden)?

**Comment by AnirudhDagar** [comment]
Yeah I'll try to investigate that within d2l-book and see if it's possible to fix that. Nice idea!

**Comment by AnirudhDagar** [comment]
Once this is merged, then we can make a new release and upload the latest d2l package. This would unblock https://github.com/d2l-ai/d2l-zh/pull/1095 and CI for d2l-zh should be running then.

**Review by astonzhang — APPROVED** [review]
